addons/mrp_subcontracting/models/stock_picking.py

```diff
@@ -73,12 +73,18 @@ def _action_done(self):
             len_amounts = len(amounts)
             # _split_production can set the qty_done, but not split it.
             # Remove the qty_done potentially set by a previous split to prevent any issue.
-            production.move_line_raw_ids.filtered(lambda l: l.state == 'assigned').write({'qty_done': 0})
+            production.move_line_raw_ids.filtered(lambda sml: sml.state != 'done').write({'qty_done': 0})
             productions = production._split_productions({production: amounts}, set_consumed_qty=True)
             for production, move_line in zip(productions, move.move_line_ids):
                 if move_line.lot_id:
                     production.lot_producing_id = move_line.lot_id
                 production.qty_producing = production.product_qty
+                # set_consumed_qty=True in _split_productions may not have assigned all raw_move_ids if the initial
+                # component move was partially available.
+                for raw_move in production.move_raw_ids:
+                    if float_is_zero(raw_move.quantity_done, precision_rounding=raw_move.product_uom.rounding)\
+                            and raw_move.product_id.tracking == "none":
+                        raw_move._set_quantity_done(raw_move.product_uom_qty)
```

Comment:
you can use `_set_quantity_producing()` like we do here

https://github.com/odoo/odoo/blob/862730899e0296e2718fae6ebca23cb77ad1178a/addons/mrp_subcontracting/models/stock_move.py#L108-L116

Reply:
Will do, it works well, but it will be applied on all raw moves instead of the ones that need it.